pyrokinetics/kinetics/jetto.py

```diff
@@ -167,6 +167,37 @@ def read_from_file(
                 rho=rho_func,
             )
 
+        # correct for the fast alpha temperature
+        ion_total_pressure = kinetics_data["PRI"][time_index, :] / elementary_charge
+        slow_pressure = np.zeros(len(psi_n))
+        for species in possible_species:
+            if not species["jetto_name"] == "NALF":
+                slow_pressure += (
+                    kinetics_data["TI"][time_index, :]
+                    * kinetics_data[species["jetto_name"]][time_index, :]
+                )
+
+        fast_pressure = ion_total_pressure - slow_pressure
+        fast_temp_data = (
+            (ion_total_pressure - slow_pressure) / kinetics_data["NALF"][time_index, :]
+        ) * units.eV
+
+        fast_temp_func = UnitSpline(psi_n, fast_temp_data)
+
+        np.savetxt("fast_pressure", fast_pressure)
+        np.savetxt("nalf", kinetics_data["NALF"][time_index, :])
+
+        result["alphas"] = Species(
```

Comment:
@bpatel2107  @LiamPattinson  Is there a way to update just the temperature? (Please ignore the error in populating the fields.)

Reply:
The alpha pressure can be taken from `2/3 * WALD` from the `jsp` file. `WALD` is the energy density of the alphas and the `2/3` converts it from energy density to a pressure

Reply:
Oh very helpful!  Hopefully the two are consistent.   :)

Reply:
I would define a `fast_ion_temp_func` from `WALD` and then when creating the fast ion `Species` object have a check for whether or not to use the thermal temp or fast ion temp 

i.e. something like
```python
thermal_ion_temp_func = kinetics_data["TI" ]
fast_ion_temp_func = 2/3 * kinetics_data["WALD" ] / kinetics_data["NALF"]
if species_name == "alpha":
    temp_func = fast_ion_temp_func
else:   
    temp_func = thermal_ion_temp_func
   
result[species["species_name"]] = Species(
    species_type=species["species_name"],
    charge=species["charge"],
    mass=species["mass"],
    dens=density_func,
    temp=temp_func,
    rot=rotation_func,
    ang=omega_func,
    rho=rho_func,
)
```

Reply:
Thanks for this.  What is the rationale behind checking for thermal ion temp for fast species?

Reply:
Right now we iterate through `possible_species` which is a dictionary of all the ion species with all the data needed to create a `Species` object. This just sets it properly when looping through `possible_species`